crates/miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -562,6 +583,57 @@ export.get_map_item
     # => [VALUE]
 end
 
+#! Returns the VALUE located under the specified KEY within the map contained in the given
+#! account storage slot at its initial state (beginning of transaction).
+#!
+#! Inputs:  [index, KEY]
+#! Outputs: [INIT_VALUE]
+#!
+#! Note:
+#! - We assume that index has been validated and is within bounds.
+#!
+#! Where:
+#! - index is the index of the storage slot that contains the map root.
+#! - INIT_VALUE is the initial value of the map item at KEY at the beginning of the transaction.
+#!
+#! Panics if:
+#! - the requested storage slot type is not map.
+export.get_map_item_init
```

Comment:
Similarly, here, I think it would be worth deduplicating the code. E.g. I think a setup (roughly) like this should work:

```
export.get_map_item_init
    exec.get_item_init
    exec.get_map_item_raw
end

export.get_map_item
    exec.get_item
    exec.get_map_item_raw
end

#! Inputs:  [index, KEY, ROOT]
#! Outputs: [VALUE]
proc.get_map_item_raw
    # check storage slot type
    # hash map key
    # smt::get
    # ...
end
```